src/reducers/utils.js

```diff
@@ -0,0 +1,50 @@
+const checkItemExist = (list, productId) => {
```

Comment:
Again great job on separating all these functions into utlis. 

This code can be simplified to ` list.find(({id,status})=>id=== productId && status!==="deleted")`. Check what find does here https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/find